src/Sema.zig

```diff
@@ -27734,8 +27749,7 @@ fn cmpNumeric(
             }
         }
         if (rhs_is_float) {
-            var bigint_space: Value.BigIntSpace = undefined;
-            var bigint = try rhs_val.toBigInt(&bigint_space, target).toManaged(sema.gpa);
+            var bigint = try float128IntPartToBigInt(sema.gpa, rhs_val.toFloat(f128));
```

Comment:
Looking at this code fresh, I'm not sure why we need to construct this huge bigint in the first place.

If the float has a fractional part, the coercion to an integer later will fail anyway, right?